tests/php/test_class.jetpack.php

```diff
@@ -783,6 +790,77 @@ function get_file_url_for_environment_data_provider() {
 		);
 	}
 
+	/**
+	 * Tests verify_xml_rpc_signature();
+	 *
+	 * @author obenland
+	 * @covers ::verify_xml_rpc_signature
+	 */
+	function test_verify_xml_rpc_signature() {
+		$jetpack = Jetpack::init();
+		$user_id = $this->factory->user->create( array(
+			'role'            => 'administrator',
+			'user_registered' => '1994-01-01 00:00:00',
+		) );
+		wp_set_current_user( $user_id );
+
+		$nonce     = wp_create_nonce();
+		$timestamp = time();
+		$token     = sprintf( 'token:%d:%d', JETPACK__API_VERSION, $user_id );
+
+		$_SERVER['REQUEST_URI'] = '/endpoint';
+		$jetpack_signature = new Jetpack_Signature( 'token.1', (int) Jetpack_Options::get_option( 'time_diff' ) );
+		$signature         = $jetpack_signature->sign_current_request( array(
+			'nonce'     => $nonce,
+			'timestamp' => $timestamp,
+			'token'     => $token,
+		) );
+
+		// No signature.
+		$this->assertFalse( $jetpack->verify_xml_rpc_signature() );
```

Comment:
Yay tests! ✨

Optional, but I would recommend breaking out cases to named tests. When things break folks have a better idea of what fails without having to run this, and can provide additional context. 

One thing to verify is if the old code passes the expected test cases here, in addition to the new ones, if you haven't checked this already.

Reply:
Good point! In 02477e9 I added context to assertions to help with that.

Also verified the old code passes the unit test (with the exception of parameter-only tests of course). Generally this test should be expanded in the future to cover other and more specific use cases.